d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperServer.java

```diff
@@ -131,7 +147,6 @@ public void onSuccess(None none)
           info(_log, sb);
         }
         _store.put(clusterName, new UriProperties(clusterName, partitionDesc, myUriSpecificProperties), callback);
```

Comment:
we want to do the Kafka announce as close to the ZK announcement so that we can compare the propagation time between INDIS control plane and ZK-architecture fairly. So the announcement should happen either right before this line.

Reply:
I thought we had discussed this previously -- didn't we decide that we want the Indis announce separate from the ZK announcement callback? I had originally had the announcement next to the store.put, but moved it out because I thinkthe `store.put` is in a callback. 

What do you think? I feel like making the ZK and Kafka announcements independent may be more important than keeping the logic close together, which means the Kafka announcement has to be either before or after the callback

I could also move it from before to after, if that would help!

Reply:
Discussed offline with @bohhyang and we agreed that keeping the Kafka announcement independent from the ZK announcement is more important

I have added a comment mentioning why this code is separate from the ZK announcement so future readers will know